include/bluetooth/gatt.h

```diff
@@ -918,6 +918,28 @@ struct bt_gatt_indicate_params {
 int bt_gatt_indicate(struct bt_conn *conn,
 		     struct bt_gatt_indicate_params *params);
 
+
+/** @brief Check if connections have subscribed to attribute
+ *
+ *  Check if connection has subscribed to attribute value change. If connection
+ *  is NULL then check if any connection has subscribed.
```

Comment:
Hmm accepting NULL may create the anti-pattern of call bt_gatt_subscribed followed by bt_gatt_notify, etc, which will check everything again, for non-NULL that should be alright since it notify regardless.

Btw, the application can maintain a flag with gets updates by the cfg_changed, that is how we normally track the state in our samples which is a lot faster than going thru the attribute user_data to inspect it.

Reply:
Should we remove the NULL functionality then?

I don't really understand the flag thing. The problem with cfg_changed is that there is no conn object given in the callback. So cfg_changed only tells you that any of the connections you have wrote that value. Our samples doesn't work properly for multiple connections.

Reply:
Well perhaps we have a bug then, notifying with NULL should do the job of finding all the connections which had enabled notifications, so yes Id remove the NULL and make it only check with a conn set.

Btw the cfg_write would give the conn, though that means you would have to maintain the state per conn which is another anti-pattern since the stack is already doing that.

Reply:
Ok. I will remove the NULL conn behaviour. But otherwise this is a good approach? Going through the attribute user_data seems to be the correct state to inspect for the stack.

Reply:
Yep, if you really want to notify on a per conn that should be alright.

Reply:
Updated.